pkg/agent/controller/networkpolicy/networkpolicy_controller.go

```diff
@@ -100,6 +101,129 @@ func (c *Controller) GetAppliedToGroupNum() int {
 	return c.ruleCache.GetAppliedToGroupNum()
 }
 
+func ipNetConverter(ipNets ...v1beta1.IPNet) (ret []networking.IPNet) {
+	for _, ipNet := range ipNets {
+		ret = append(ret, networking.IPNet{
+			IP:           networking.IPAddress(ipNet.IP),
+			PrefixLength: ipNet.PrefixLength,
+		})
+	}
+	return ret
+}
+
+func ipBlocksConverter(blocks []v1beta1.IPBlock) (ret []networking.IPBlock) {
+	for _, block := range blocks {
+		ret = append(ret, networking.IPBlock{
+			CIDR:   ipNetConverter(block.CIDR)[0],
+			Except: ipNetConverter(block.Except...),
+		})
+	}
+	return ret
+}
+
+func portsConverter(ports ...v1beta1.NamedPort) (ret []networking.NamedPort) {
+	for _, port := range ports {
+		ret = append(ret, networking.NamedPort{
+			Port:     port.Port,
+			Name:     port.Name,
+			Protocol: networking.Protocol(port.Protocol),
+		})
+	}
+	return ret
+}
+
+func serviceConverter(services ...v1beta1.Service) (ret []networking.Service) {
+	for _, service := range services {
+		protocol := networking.Protocol(*service.Protocol)
+		ret = append(ret, networking.Service{
+			Protocol: &protocol,
+			Port:     service.Port,
+		})
+	}
+	return ret
+}
```

Comment:
External consumers should use versioned type only, no need to do above conversion. internal type is mutable, so could break compatibility.

Reply:
Outdated.

Reply:
what do you mean by "outdated"?

Reply:
Sorry, I closed too quick. Updated, PTAL.